agent/src/main/scala/za/co/absa/atum/agent/dispatcher/CapturingDispatcher.scala

```diff
@@ -19,121 +19,140 @@ package za.co.absa.atum.agent.dispatcher
 import com.typesafe.config.Config
 import za.co.absa.atum.model.dto._
 
-import java.util
-import scala.jdk.CollectionConverters._
+import java.util.concurrent.atomic.AtomicReference
+import java.util.function.UnaryOperator
+import scala.collection.immutable.Queue
 
 /**
  *  This dispatcher instead of sending data captures them and stores them in memory.
- *  @param checkpointsLimit maximal amount of checkpoints to be stored for each partition.
- *                   If the limit is reached, the oldest checkpoint is removed.
- *                   Value 0 removes this limit.
+ *  @param config: Config to be used to create the dispatcher. Keys:
+ *           capture-limit - maximal amount of dispatch captures to store.
  */
-class CapturingDispatcher(checkpointsLimit: Int) extends Dispatcher {
+class CapturingDispatcher(config: Config) extends Dispatcher(config) {
   import CapturingDispatcher._
 
-  private val ts = util.Collections.synchronizedSortedMap(new util.TreeMap[String, PartitionedData]())
+
+  val captureLimit: Int = config.getInt(CheckpointLimitKey)
 
   /**
-   *  This method is used to ensure the server knows the given partitioning.
-   *  As a response the `AtumContext` is fetched from the server.
+   * This method is used to clear all captured data.
+   */
+  def clear(): Queue[CapturedCall] = {
+    capturesRef.updateAndGet((_: Queue[CapturedCall]) => Queue.empty)
+    captures
```

Comment:
I assume this is a leftover from the code before the last change. Because the `updateAndGet` updates the state and returns the new state, which is an empty queue. 

The additional `captures` call would call `capturesRef.get()`, which might differ from the expected empty queue. This is because a separate thread might modify the queue during the clear operation, which will alter the empty queue right after `updateAndGet` and before `captures`.

So, if this is intentional, please state that in the docstring and explain the intention. If this is unintended, please remove the `captures` call at the end because this might cause some sneaky, undefined behavior.

Reply:
Not a leftover, but not-though through feature. It's not worth the effort. Returning, `Unit` is plenty enough.